core/arch/riscv64/riscv64.asm

```diff
@@ -208,16 +208,51 @@ ADDRTAKEN_LABEL(safe_read_asm_recover:)
  */
         DECLARE_EXPORTED_FUNC(dr_try_start)
 GLOBAL_LABEL(dr_try_start:)
-        addi      ARG1, ARG1, TRY_CXT_SETJMP_OFFS
+        addi     ARG1, ARG1, TRY_CXT_SETJMP_OFFS
         j        GLOBAL_REF(dr_setjmp)
         END_FUNC(dr_try_start)
 
-/*
+/* We save only callee-saved regites: SP, x8/fp, x9, x18-x27, f8-9, f18-27:
+ * a total of 25 reg_t (64-bit) slots. See definition of dr_jmp_buf_t.
+ *
  * int dr_setjmp(dr_jmp_buf_t *buf);
  */
         DECLARE_FUNC(dr_setjmp)
 GLOBAL_LABEL(dr_setjmp:)
-/* FIXME i#3544: Not implemented */
+        sd       x18, 0 (ARG1)
+        sd       x19, 8 (ARG1)
+        sd       x20, 16 (ARG1)
+        sd       x21, 24 (ARG1)
+        sd       x22, 32 (ARG1)
+        sd       x23, 40 (ARG1)
+        sd       x24, 48 (ARG1)
+        sd       x25, 56 (ARG1)
+        sd       x26, 64 (ARG1)
+        sd       x27, 72 (ARG1)
+        mv       ARG7, sp
```

Comment:
Curious: here we're using ARG7 as a temporary reg, right? Could we have used any of the many tx temporary regs provided by riscv?

Reply:
Done

Reply:
Fixed https://github.com/DynamoRIO/dynamorio/pull/6071/commits/3859e2075255495463f4d09652e54771306f0a75